libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1230,6 +1246,29 @@ string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::cleanUpDynamicByteArrayEndSlotsFunction(ArrayType const& _type)
+{
+	solAssert(_type.isByteArray(), "");
+	solAssert(_type.isDynamicallySized(), "");
+
+	string functionName = "clean_up_byte_array_end_slots_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&](vector<string>& _args, vector<string>&) {
+		_args = {"array", "len", "pos"};
+		return Whiskers(R"(
+			if gt(len, 31) {
+				let dataArea := <dataLocation>(array)
+				let deleteStart := add(dataArea, div(<roundUp>(pos), 32))
+				if lt(pos, 32) { deleteStart := dataArea }
```

Comment:
Strictly, this is not documented, but I think it's fine. Maybe we should document "just look at the code"? :)

Reply:
Actually: This detail is really important if someone wants to use his helper for shortening a byte array...

Reply:
I think it at least warrants a comment. At first I could not understand why this special case is here but then I realized that if `pos` is < 32, we're assuming that the data has already been moved to the slot that stores length and we need to clear the beginning of the data area too. It might have been more obvious when this was a part of the bigger function but now this lacks context.

Reply:
I added a comment before this line